ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/solution_expose.py

```diff
@@ -75,7 +75,11 @@ def domain_1(self):
         free_to_use = False
         if "FreeTFT" == self.solution_currency:
             free_to_use = True
-        self.gateways = {g.node_id: g for g in j.sal.zosv2._explorer.gateway.list() if g.free_to_use == free_to_use}
+        self.gateways = {
+            g.node_id: g
+            for g in j.sal.zosv2._explorer.gateway.list()
+            if g.free_to_use == free_to_use and j.sal.reservation_chatflow.node_is_up(g.updated)
```

Comment:
why not use https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/sal/zosv2/node_finder.py#L13 ?